uwblueprint/supportive-housing
# Frontend Filtering

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description
* modifies filtering in the backend to allow us to filter by any fields instead of all fields at once
* doesn't including searching by resident bc the resident API changes are not merged yet
* currently we directly filter by employee IDs and attention To user IDs because we'll be able to convert name -> ID once the log records component takes info from the API
* the current implementation isn't completely accurate to the designs because it's more reliant on the design system being implemented + some other small nits i couldn't figure out on a short term 😅 



## Steps to test
1. add dummy data into your database table:
`INSERT INTO log_records (employee_id, resident_first_name, resident_last_name, datetime, flagged, attn_to, note, tags, building) VALUES (1, 'John', 'Doe', '2023-04-10 12:34:56', false, 2, 'This is a note', ARRAY['tag1', 'tag2'], 'Building C');`
`INSERT INTO log_records (employee_id, resident_first_name, resident_last_name, datetime, flagged, attn_to, note, tags, building) VALUES (1, 'John', 'Doe', '2023-04-20 12:34:56', false, 2, 'This is a note', ARRAY['tag1', 'tag2'], 'Building A');`
`INSERT INTO log_records (employee_id, resident_first_name, resident_last_name, datetime, flagged, attn_to, note, tags, building) VALUES (1, 'John', 'Doe', '2023-04-20 12:34:56', false, NULL, 'This is a note', ARRAY['tag1', 'tag2'], 'Building A');`
2. type in what you want to filter. Examples include: 
- attention to: 2 (should return the first two entries)
- start date: 2023-04-09, end date: 2023-04-11 (should return the first entry)
- employee: 1 (should return all three)
3. the console.log in LogRecords will display what logs are returned through the filtering


## What should reviewers focus on?
* 


## Checklist
